code/controllers/subsystem/chat.dm

```diff
@@ -20,16 +20,16 @@ SUBSYSTEM_DEF(chat)
 	var/list/client_to_sequence_number = list()
 
 /datum/controller/subsystem/chat/proc/generate_payload(client/target, message_data)
-	var/sequence = client_to_sequence_number[target.ckey]
-	client_to_sequence_number[target.ckey] += 1
+	var/sequence = client_to_sequence_number[target.ckey]++
 
 	var/datum/chat_payload/payload = new
 	payload.sequence = sequence
 	payload.content = message_data
 
-	if(!(target.ckey in client_to_reliability_history))
-		client_to_reliability_history[target.ckey] = list()
 	var/list/client_history = client_to_reliability_history[target.ckey]
+	if(!islist(client_history))
+		client_history = (client_to_reliability_history[target.ckey] = list())
```

Comment:
also, how common is this case?

Reply:
once per-client for the first message.

a null check would work here, but this is a key piece of a core framework (this is the key way the server communicates with admins and players) so the extra reliability of doing an islist() seemed free. (`islist()` uses byond's `valuetype` field so its almost the same cost as `isnull()`)

Reply:
If it's that rare do you mind making it lot uh `thing1 = thing2 = value`? I'd like to keep that pattern as locked away as possible cause it kinda sucks